src/openms/source/ANALYSIS/OPENSWATH/MRMDecoy.cpp

```diff
@@ -379,6 +378,27 @@ namespace OpenMS
     return false;
   }
 
+  String MRMDecoy::getModifiedPeptideSequence_(const OpenMS::TargetedExperiment::Peptide& pep) const
+  {
+          String full_peptide_name = "";
+          for (int loc = -1; loc <= (int)pep.sequence.size(); loc++)
+          {
+            if (loc > -1 && loc < (int)pep.sequence.size())
+            {
+              full_peptide_name += pep.sequence[loc];
+            }
+            // C-terminal and N-terminal modifications may be at positions -1 or pep.sequence
+            for (Size modloc = 0; modloc < pep.mods.size(); modloc++)
```

Comment:
Having this inner loop is potentially very inefficient. How often is this function called?

Reply:
This is called once per peptide sequence so it might be called a lot. However the inner loop should only be called maybe maximum of 3 times because theoretically there are not too many modifications on a peptide. This code block is actually taken from OpenMS tsv writer. https://github.com/OpenMS/OpenMS/blob/c2555c3278de34a5f261df9a0915f42a8bfada3c/src/openms/source/ANALYSIS/OPENSWATH/OpenSwathTSVWriter.cpp#L145-L160
however this is probably not the most efficient way. Basically here I am just looking for a data structure which can hold modified peptide sequences to check for equality.

Reply:
Running with a big library with over 2000000 it generates decoys in a reasonable time for me (e.g. 20 sec) so I think this implementation is good enough for now